pkg/apis/workspaces/v1alpha2/components.go

```diff
@@ -29,6 +29,8 @@ type Component struct {
 	// Mandatory name that allows referencing the component
 	// from other elements (such as commands) or from an external
 	// devfile that may reference this component through a parent or a plugin.
+	// +kubebuilder:validation:Pattern=^([a-z0-9]([-a-z0-9]*[a-z0-9]))?$
+	// +kubebuilder:validation:MaxLength=63
```

Comment:
MinLength should be 1? It should error out if there is an empty string for some reason

Reply:
Probably doesnt matter if name/id is mandatory